benchmarks/definitions/nashpobench.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from pathlib import Path
+import numpy as np
+from random import shuffle
+
+from sklearn.ensemble import RandomForestRegressor
+
+from syne_tune.search_space import choice, randint, uniform, loguniform
+
+from blackbox_repository.conversion_scripts.scripts.fcnet_import import \
+    METRIC_ELAPSED_TIME, METRIC_VALID_LOSS, RESOURCE_ATTR, BLACKBOX_NAME
+
+
+class SubsamplingSurrogate(object):
+    """
+    Fits a random forest to project the discrete space back to the original space with continuous and integer
+    hyperparameters. The entire dataset is quite large (for each config we have 4 seeds, 100 epochs, 4 metrics). To
+    reduce building time, we subsample the original dataset first before feeding it to the Random Forest.
+
+    """
+    def __init__(self, ratio=0.01):
+        self.ratio = ratio
+
+    def fit(self, X, y):
```

Comment:
Could we just use `max_samples` argument of RandomForestRegressor? It is not clear why we want to have a class.

Reply:
good catch, didn't know this argument